vere/sist.c

```diff
@@ -982,45 +820,112 @@ _sist_rest()
       u3_lo_bail();
     }
   }
-  else {
-    u3_noun rou = roe;
-    c3_w    xno_w;
 
-    //  Execute the fscking things.  This is pretty much certain to crash.
+
+  //  Read and execute the fscking things. This is pretty much certain to crash.
+  //
+  uL(fprintf(uH, "rest: replaying through event %" PRIu64 "\n", las_d));
+  fprintf(uH, "---------------- playback starting----------------\n");
```

Comment:
This playback used to only run if `u3_nul != roe`, ie, if there were events in the log but not in the snapshot. Why is it unconditional now?

Reply:
It's not unconditional anymore (looks like @ixv fixed this). Now instead of loading the whole list of events since the last checkpoint (roe) into memory and checking for null, we check whether our event pointer `cur_d` is at the end of the file. If it is, we assert the event log and checkpoint ended at the same event.

Reply:
We seem to always have one event to replay now. Previously we would only have replay events if the process crashed, was killed with ctrl-z, or if we removed the checkpoint. This isn't a huge problem, but it's kind of annoying.

Reply:
Oh, that's what's going on! I knew I was annoyed, but wasn't sure why. I'll see if I can fix this